lib/common/strings.c

```diff
@@ -148,6 +150,120 @@ crm_parse_int(const char *text, const char *default_text)
     return (int) result;
 }
 
+/*!
+ * \internal
+ * \brief Scan a double-precision floating-point value from a string
+ *
+ * \param[in]      text         The string to parse
+ * \param[out]     result       Parsed value on success, or -1 on error
+ * \param[in]      default_text Default string to parse if \p text is
+ *                              \c NULL
+ * \param[out]     end_text     If not \c NULL, where to store a pointer
+ *                              to the position immediately after the
+ *                              value
+ *
+ * \return Standard Pacemaker return code (\c pcmk_rc_ok on success,
+ *         \c EINVAL on failed string conversion due to invalid input,
+ *         \c EOVERFLOW on arithmetic overflow, \c pcmk_rc_underflow
+ *         on arithmetic underflow, or \c errno from \c strtod() on
+ *         other parse errors)
+ */
+int
+pcmk__scan_double(const char *text, double *result, const char *default_text,
+                  char **end_text)
+{
+    int rc = pcmk_rc_ok;
+    char *local_end_text = NULL;
+
+    CRM_ASSERT(result != NULL);
+    *result = -1.0;
```

Comment:
FYI there is a use case for result==NULL ... the caller could just want to check whether a string would successfully parse or not (checking the validity of command-line arguments or option values is an example). But don't change this, we don't have a use case for it now and it's internal API so we can change it later if needed.

Reply:
I actually considered that, and decided "screw it" for the reasons you mentioned.